transport_analysis/velocityautocorr.py

```diff
@@ -262,3 +261,68 @@ def plot_vacf(self, start=0, stop=0, step=1):
             self.times[start:stop:step],
             self.results.timeseries[start:stop:step],
         )
+
+    def sd(self, start=0, stop=0, step=1):
+        """
+        Returns a self-diffusivity value using ``scipy.integrate.trapezoid``.
+        Analysis must be run prior to computing self-diffusivity.
+
+        Parameters
+        ----------
+        start : Optional[int]
+            The first frame of ``self.results.timeseries``
+            used for the calculation.
+        stop : Optional[int]
+            The frame of ``self.results.timeseries`` to stop at
+            for the calculation, non-inclusive.
+        step : Optional[int]
+            Number of frames to skip between each frame used
+            for the calculation.
+
+        Returns
+        -------
+        `numpy.float64`
+            The calculated self-diffusivity value for the analysis.
+        """
+        stop = self.n_frames if stop == 0 else stop
+
+        return (
+            integrate.trapezoid(
+                self.results.timeseries[start:stop:step],
+                self.times[start:stop:step],
+            )
+            / self.dim_fac
+        )
+
+    def sd_odd(self, start=0, stop=0, step=1):
```

Comment:
We will probably remove this IMHO as trapeziod or Gquad is the standard. Can keep it around for testing but I would make it a private method with a `_` prefix

Reply:
I thought Simpson's rule was better when appropriate? Here's a link to my source: https://math.dartmouth.edu/~m3cod/klbookLectures/406unit/trap.pdf

Is keeping the standard more important than having an option for better accuracy?

Reply:
Need for an odd number of samples is a bit of a pain.

Reply:
I feel having the option available is better than not providing it, but I am no expert. Is it really preferable to remove it rather than just make it clear that trapezoid is the standard/default?

Reply:
In general you want to aim for the lowest possible API surface so that people don't get confused and do the wrong thing. 

The idea being you should plan for users to get frustrated as soon as the first thing goes wrong, meaning you should take away the ability for them to annoy themselves if that makes sense.

Reply:
If users do the wrong thing (use the method with an even number of samples), `scipy.integrate.simpson` actually handles it well by default using the equations in Cartwright's paper for the last interval. On second thought, considering that it is pretty much the same as trapezoid but better, what are your thoughts on using it as the default instead?

Source: https://docs.scipy.org/doc/scipy/reference/generated/scipy.integrate.simpson.html